src/main/sensors/barometer.c

```diff
@@ -411,12 +411,16 @@ uint32_t baroUpdate(timeUs_t currentTimeUs)
             if (baro.dev.read_ut(&baro.dev)) {
                 state = BARO_STATE_TEMPERATURE_SAMPLE;
             }
+            // No action was taken as the read has not completed
+            schedulerIgnoreTaskStateTime();
             break;
 
         case BARO_STATE_TEMPERATURE_SAMPLE:
             if (baro.dev.get_ut(&baro.dev)) {
                 state = BARO_STATE_PRESSURE_START;
             }
+            // No action was taken as the read has not completed
+            schedulerIgnoreTaskStateTime();
```

Comment:
Should this be in else clause?

Reply:
And that, my friend, is why we have reviews! Thank you. Fixed.